hidimstat/desparsified_lasso.py

```diff
@@ -85,68 +139,246 @@ def desparsified_lasso_confint(X, y, confidence=0.95, max_iter=5000,
+        updates are smaller than `tol`, the optimization code checks the
+        dual gap for optimality and continues until it is smaller than `tol`.
+
+    residual_method : string, optional (default='lasso')
+        The method for the computind the residuals of the Nodewise Lasso.
+        Currently the only method available is 'lasso'.
+
+    c : float, optional (default=0.01)
+        Only used if method='lasso'. Then alpha = c * alpha_max.
+
+    noise_method : string, optional (default='simple')
+        If 'simple', the correlation matrix is estimated by taking the
+        median of the correlation between two consecutive time steps
+        and the noise standard deviation for each time step is estimated
+        by taking the median of the standard deviations for every time step.
+        If 'AR', the order of the AR model is given by `order` and the
+        Yule-Walker method is used to estimate the covariance matrix.
+
+    order : int, optional (default=1)
+        If `method=AR`, `order` gives the order of the estimated autoregressive
+        model. `order` must be smaller than the number of time steps.
+
+    n_jobs : int or None, optional (default=1)
+        Number of CPUs to use during the Nodewise Lasso.
+
+    memory : str or joblib.Memory object, optional (default=None)
+        Used to cache the output of the computation of the Nodewise Lasso.
+        By default, no caching is done. If a string is given, it is the path
+        to the caching directory.
+
+    verbose: int, optional (default=1)
+        The verbosity level: if non zero, progress messages are printed
+        when computing the Nodewise Lasso in parralel.
+        The frequency of the messages increases with the verbosity level.
+
+    Returns
+    -------
+    beta_hat : ndarray, shape (n_features, n_times)
+        Estimated parameter matrix.
+
+    sf : ndarray, shape (n_features,)
+        Survival function values of every feature
+
+    sf_corr : ndarray, shape (n_features,)
+        Corrected survival function values of every feature
+
+    cdf : ndarray, shape (n_features,)
+        Cumulative distribution function values of every feature
+
+    cdf_corr : ndarray, shape (n_features,)
+        Corrected cumulative distribution function values of every feature
+
+    References
+    ----------
+    .. [1] Chevalier, J. A., Gramfort, A., Salmon, J., & Thirion, B. (2020).
+           Statistical control for spatio-temporal MEG/EEG source imaging with
+           desparsified multi-task Lasso. In NeurIPS 2020-34h Conference on
+           Neural Information Processing Systems.
+    """
+
+    X = np.asarray(X)
+
+    n_samples, n_features = X.shape
+    n_times = Y.shape[1]
+
+    memory = check_memory(memory)
+
+    if cov is not None and cov.shape != (n_times, n_times):
+        raise ValueError(f'Shape of cov should be ({n_times}, {n_times})')
+
+    if normalize:
+
+        Y = Y - np.mean(Y)
```

Comment:
```suggestion
        Y = Y - np.mean(Y, axis=0)
```
did you consider this option?

Reply:
Yes and I believe it is not a good idea since it would modify the temporal structure. WDYT ?

Reply:
agreed